core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ApplyTableScanRedirection.java

```diff
@@ -86,15 +87,18 @@ public Result apply(TableScanNode scanNode, Captures captures, Context context)
 
         Redirection redirection = getOnlyElement(redirections);
         CatalogSchemaTableName destinationTable = redirection.getDestinationTable();
+        TableMetadata tableMetadata = metadata.getTableMetadata(context.getSession(), scanNode.getTable());
```

Comment:
It needs to be in this commit because the modification to MockConnectorTableHandle#equals requires this change to fix test failures.
The other commit is just taking advantage of the presence of sourceTable